### PR TITLE
Ensure consistent placement of error notifications on list pages

### PR DESCRIPTION
**Pull request by AlanGreene**
# Changes

Move responsibility for display of error notifications to the common
`ResourceList` and `ListPageLayout` components to ensure consistency
across all resource types.

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Includes [tests](https://github.com/tektoncd/community/blob/main/standards.md#principles) (if functionality changed/added)
- [ ] Includes [docs](https://github.com/tektoncd/community/blob/main/standards.md#principles) (if user facing)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/main/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/dashboard/blob/main/CONTRIBUTING.md)
for more details._

### CODE REVIEW

**Comment by AlanGreene** [comment]
closed and reopened to trigger EasyCLA check

**Review by steveodonovan — COMMENTED** [review]
One nit and a qq, I see we handle delete errors in those get error functions, do we want to consider all actions there ? (rerun and cancel) Throw them all under an action error in the state and reuse that

File: src/containers/ClusterTriggerBindings/ClusterTriggerBindings.js [hunk 1]
```diff
@@ -55,6 +49,15 @@ export /* istanbul ignore next */ class ClusterTriggerBindings extends Component
     }
   }
 
+  getError() {
+    const { error } = this.props;
+    if (error) {
+      return { error };
```
Comment (steveodonovan) [inline src/containers/ClusterTriggerBindings/ClusterTriggerBindings.js:55]: Looks like my nit didn't save - is error always just a message here ? if so should we call it message given the object gets passed around as error

Comment (AlanGreene) [reply src/containers/ClusterTriggerBindings/ClusterTriggerBindings.js:55]: Today it is, but that won't be the case for long. With https://github.com/tektoncd/dashboard/issues/1842 we'll have `Error` objects or similar with more information. Trying to get things cleaned up before that so it's a bit easier to manage the switch.

**Review by steveodonovan — APPROVED** [review]
/approve
/lgtm

**Pull request merged**